src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/WalletController.cs

```diff
@@ -44,4 +45,57 @@ public async Task<ActionResult<ResultList<Claim>>> GetClaims([FromServices] IUni
 
         return new ResultList<Claim> { Result = claims.Select(c => c.MapToV1()).ToArray() };
     }
+
+    [HttpGet]
+    [Route("v1/aggregate-certificates")]
+    [Produces("application/json")]
+    public async Task<ActionResult<ResultList<AggregationResult>>> AggregateCertificates([FromServices] IUnitOfWork unitOfWork, [FromQuery] TimeAggregate timeAggregate, [FromQuery] SliceState state, [FromQuery] long? start, [FromQuery] long? end, [FromQuery] CertificateType? type)
+    {
+        var owner = User.GetSubject();
+
+        var certificates = await unitOfWork.CertificateRepository.GetAllOwnedCertificates(owner, new CertificatesFilter(state)
+        {
+            Start = start != null ? DateTimeOffset.FromUnixTimeSeconds(start.Value) : null,
+            End = end != null ? DateTimeOffset.FromUnixTimeSeconds(end.Value) : null,
+            Type = type != null ? Map(type.Value) : null
+        });
+
+        if (timeAggregate == TimeAggregate.Day)
+        {
+            return new ResultList<AggregationResult> { Result = GroupCertificatesHelper.GroupByDay(certificates).ToArray() };
+        }
+
+        throw new NotImplementedException($"timeAggregate value {timeAggregate} has not been implemented.");
+    }
+
+    [HttpGet]
+    [Route("v1/aggregate-claims")]
+    [Produces("application/json")]
+    public async Task<ActionResult<ResultList<AggregationResult>>> AggregateClaims([FromServices] IUnitOfWork unitOfWork, [FromQuery] TimeAggregate timeAggregate, [FromQuery] long? start, [FromQuery] long? end)
+    {
+        var owner = User.GetSubject();
+
+        var claims = await unitOfWork.CertificateRepository.GetClaims(owner, new ClaimFilter
+        {
+            Start = start != null ? DateTimeOffset.FromUnixTimeSeconds(start.Value) : null,
+            End = end != null ? DateTimeOffset.FromUnixTimeSeconds(end.Value) : null,
+        });
+
+        if (timeAggregate == TimeAggregate.Day)
+        {
+            return new ResultList<AggregationResult> { Result = GroupClaimsHelper.GroupByDay(claims).ToArray() };
+        }
+
+        throw new NotImplementedException($"timeAggregate value {timeAggregate} has not been implemented.");
+    }
+
+    private static GranularCertificateType Map(CertificateType type)
+    {
+        return type switch
+        {
+            CertificateType.Production => GranularCertificateType.Production,
+            CertificateType.Consumption => GranularCertificateType.Consumption,
+            _ => throw new ArgumentException($"Unsupported certificate type {type}")
+        };
+    }
```

Comment:
We could easily move some of this to the repository/service layer.

Reply:
To what advantage?